### DISCUSSION TITLE
Wie "nRF24 Chip Status	not connected" debuggen?

### DISCUSSION THREAD

**Discussion by arne21** [tbnobody/OpenDTU#1012] [Q&A]
Ich hatte mal ein funktionierendes OpenDTU-Setup, das ich leider zerstört habe beim Versuch, es in ein zu kleines Gehäuse zu pressen (Micro-USB-Buchse am ESP32 abgebrochen).

Mit einem neuen ESP32-Modul komme ich bis zu "OpenDTU ist in meinem WLAN und ich kann auf das Web-Interface zugreifen", aber alle Daten sind 0 und in ".../info/system" -> "Radio Information" sehe ich "nRF24 Status	configured" und "nRF24 Chip Status	not connected". Im Serial Monitor (vs code) sehe ich als einzig relevant scheinenden Eintrag "Initialize Hoymiles interface... NRF: Connection error!!"

Das alte ESP32 war eins mit "D"-Pins und ohne D16, weshalb ich den VS-Code+platform_override.ini-ändern+compilieren-Weg gegangen bin. Das neue ist eins mit G-Labels und vorhandenem G16, aber ich habe wieder VS code benutzt (ohne Port-Änderung in platform_override.ini)

Was ich bereits versucht habe, ohne Änderung am Problem:
* doppelt und dreifach checken der Verkabelung + Ersetzen der Kabel
* Ersetzen des NRF24L01+ durch ein neues Modul 
* Doch mal Port 16 auf 15 + Verkabelung entsprechend ändern
* ältere OpenDTU-Version benutzen (die, mit der es vor ein paar Wochen funktioniert hat)
* AhoyDTU mit NodeMCU-PyFlasher flashen (sehe das WLAN, dass aufgebaut werden sollte, nicht) und OpenDTU aus -generic.bin flashen (sehe openDTU-... WLAN auch nicht); in beiden Fällen blinkt die LED auf dem ESP32 schnell, bei den anderen Versuchen bleibt sie nach kurzer Blinkerei nach Einschalten aus
* diverse Netzteile und Kabel-Kombinationen

Ich bin kurz davor, mir wieder ein ESP32 mit D-Pins zu bestellen. Kann ich noch etwas anderes versuchen?

**Comment by TecKevZ** [comment]
Hey Arne21,

ich habe ziemlich genau das gleiche Problem, dass einfach das Funkmodul nicht erkannt wird.
Habe auch schon die Verkabelung kontrolliert und komlett durch neue Kabel getauscht, ein neues NRF24L01+ Modul gekauft und probiert und den ESP32 resettet und neu geflasht.... Alles ohne Erfolg. Immer noch das gleiche Problem....

Hast du mittlerweile eine Lösung?

Kann es evtl. auch am ESP32 liegen, dass dort ein defekt vorliegt?

**Comment by schnickes1** [comment]
Hallo an alle,

ich habe seit einem Update auf die aktuelle Version genau das gleiche Problem wie @arne21.
Vorher lief alles einwandfrei.

**Comment by tbnobody** [comment]
> ich habe seit einem Update auf die aktuelle Version genau das gleiche Problem wie @arne21.

Welche Datei hast du geflashed? Bei vorcompilierten Versionen ist das hier zu beachten: https://github.com/tbnobody/OpenDTU/wiki/Binary-Installation#overview-of-pre-compiled-binaries

**Reply by schnickes1** [reply]
Mein Vorgehen war wie bei arne21 mit VS Code. Dort habe ich nichts explizit ausgewählt oder anders konfiguriert.
So habe ich alle Updates bisher gemacht.

**Reply by schnickes1** [reply]
Ich habe jetzt auch nochmal eine Installation per Weboberfläche (Firmware-Aktualisierung) mit dem opendtu-generic_esp32.bin durchgeführt. Gleiches Ergebnis.

Mit der opendtu-generic.bin hat es dann funktioniert. Die scheint ja aber deprecated zu sein.

**Reply by tbnobody** [reply]
Bitte nochmal lesen. Wenn du kein device Profile hast, dann bitte das generic Profile (ohne esp32) verwenden

**Reply by schnickes1** [reply]
Ich hatte mein Kommentar nochmal akualisiert.

Wie nutze ich denn die Device Profile?
Ich finde dazu keine Anleitung wie diese hochgeladen werden.

**Reply by TecKevZ** [reply]
Ich konnte das Problem bei mir beheben! 

Nach den hinweisen von @tbnobody ist mir aufgefallen, dass in der platformio.ini in der Zeile 12 der default_envs = generic_esp32 gesetzt ist! aber für generic_esp32 sind keine Pins zugeordnet?? 
Ich habe den Wert von generic_esp32 auf generic geändert und den ESP neu geflasht. Siehe da, alles funktioniert wie es soll :) 👍

**Reply by mathuieu** [reply]
Thanks @TecKevZ i'll try tonight to change the default env too !

**Reply by d4brtn** [reply]
@TecKevZ thanks for this relevant input. I flashed my ESP with VSC and did not realise that I need to flash the pre-compiled .bin files too. I chose the way to update the .bin file via OTA and now it works very well 😄

**Comment by tbnobody** [comment]
Die Dokumention der Device Profiles ist aktuell noch hier zu finden:
[https://github.com/tbnobody/OpenDTU/blob/master/docs/DeviceProfiles.md](https://github.com/tbnobody/OpenDTU/blob/master/docs/DeviceProfiles.md)

Ziel ist es die Pins usw. nur noch per Config File festzulegen und nicht mehr zu Compile Zeit. Das hat den Vorteil mit wenigen vor-kompilierten binaries eine große anzahl an Dev-Boards usw. abzudecken. Standard .bin Datei flashen, Device Profile Hochladen, fertig.

**Reply by brekemeier** [reply]
Dazu müsste die pin_mapping.json/config.json  vorhanden sein, ist sie aber nicht, nur ein *.bin. Das kann doch nicht richtig sein? Ich teste openDTU auf  verschiedenen ESPs. Die gekaufte Version mit Hardware hat bis zum ersten Update funktioniert. Eine Anleitung lag bei, aber ohne Hinweis auf die *.json.

**Comment by brekemeier** [comment]
Nach der Lesestunde bei GitHub, habe ich alle ESPs und die "config.json und pin_mapping.json" kontrolliert und konfiguriert. Die factory-bin lassen sich nicht flashen, immer "rot" im Tool. Mit der generic-*.bin kann zumindest der Host 192.168.4.1 aufgerufen werden. Beim Reboot, nach dem flashen, zeigt die Console schon Error bei der NRF01+ an. Die Default-Pins, des NRF24+, funktionieren nicht.  Bei gleicher Konfiguration arbeitet AhoyDTU sofort, mit Display und Wechselrichter.

**Comment by brekemeier** [comment]
Thanks, have a nice Update 😄

Am 27. November 2023 13:50:09 MEZ schrieb d4brtn ***@***.***>:
